dotnet/src/Agents/OpenAI/Internal/AssistantThreadActions.cs

```diff
@@ -373,26 +373,57 @@ public static async IAsyncEnumerable<StreamingChatMessageContent> InvokeStreamin
 
         // Evaluate status and process steps and messages, as encountered.
         HashSet<string> processedStepIds = [];
-        HashSet<string> messageIds = [];
-
+        Dictionary<string, RunStep?> activeMessages = [];
         ThreadRun? run = null;
+        RunStep? currentStep = null;
+
         IAsyncEnumerable<StreamingUpdate> asyncUpdates = client.CreateRunStreamingAsync(threadId, agent.Id, options, cancellationToken);
         do
         {
-            messageIds.Clear();
+            activeMessages.Clear();
 
             await foreach (StreamingUpdate update in asyncUpdates.ConfigureAwait(false))
             {
                 if (update is RunUpdate runUpdate)
                 {
                     run = runUpdate.Value;
 
-                    logger.LogOpenAIAssistantCreatedRun(nameof(InvokeAsync), run.Id, threadId);
+                    switch (runUpdate.UpdateKind)
+                    {
+                        case StreamingUpdateReason.RunCreated:
+                            logger.LogOpenAIAssistantCreatedRun(nameof(InvokeAsync), run.Id, threadId);
+                            break;
+                    }
                 }
                 else if (update is MessageContentUpdate contentUpdate)
                 {
-                    messageIds.Add(contentUpdate.MessageId);
-                    yield return GenerateStreamingMessageContent(agent.GetName(), contentUpdate);
+                    switch (contentUpdate.UpdateKind)
+                    {
+                        case StreamingUpdateReason.MessageUpdated:
+                            yield return GenerateStreamingMessageContent(agent.GetName(), contentUpdate);
+                            break;
+                    }
+                }
+                else if (update is MessageStatusUpdate statusUpdate)
```

Comment:
Should we update any of the concepts and/or tests to show how to make use of these updates?

Reply:
Yes, I considered that also.  There is some other sample work I have queued-up for this week.  Perhaps I can address in that cycle?

https://github.com/microsoft/semantic-kernel/issues/9035